app/Http/Controllers/Backend/BookController.php

```diff
@@ -54,30 +59,54 @@ public function show()
     /**
      * Show the form for editing the specified resource.
      *
+     * @param int $id id
+     *
      * @return \Illuminate\Http\Response
      */
-    public function edit()
+    public function edit($id)
     {
-        //
+        $list = Book::find($id);
+        $categories = Category::lists('name', 'id');
+        $adminuser = AdminUser::lists('username', 'id');
+        return view('admin.book.edit', compact('list', 'categories', 'adminuser'));
     }
 
     /**
      * Update the specified resource in storage.
      *
+     * @param \Illuminate\Http\Request $request request
+     * @param int                      $id      id
+     *
      * @return \Illuminate\Http\Response
      */
-    public function update()
+    public function update(BookEditRequest $request, $id)
     {
-        //
+        $data = $request->all();
+        if ($request->hasFile('image')) {
+            $img = time() . '_' .$request->file('image')->getClientOriginalName();
+            $request->file('image')->move('images/uploads/books/', $img);
```

Comment:
[rank C] I just commented for another member in your team about this folders path, please check it. And we don't want to use hardcode anymore.